### PR TITLE
feat(traffic_light_classifier): add parameters to launch file for using different tensorrt models

### PR DESCRIPTION
**Pull request by Zeysthingz**
## Description
Added some usefull parameters  to launch file for making easier to use different models on traffic light classification node.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: perception/traffic_light_classifier/src/cnn_classifier.cpp [hunk 1]
```diff
@@ -39,10 +39,14 @@ CNNClassifier::CNNClassifier(rclcpp::Node * node_ptr) : node_ptr_(node_ptr)
   input_c_ = node_ptr_->declare_parameter("input_c", 3);
   input_h_ = node_ptr_->declare_parameter("input_h", 224);
   input_w_ = node_ptr_->declare_parameter("input_w", 224);
+  mean_ = node_ptr_->declare_parameter("mean", std::vector<double>({0.242, 0.193, 0.201}));
+  std_ = node_ptr_->declare_parameter("std", std::vector<double>({1.0, 1.0, 1.0}));
+  input_name_ = node_ptr_->declare_parameter("input_name", std::string("input_0"));
```
Comment (yukkysaito) [inline perception/traffic_light_classifier/src/cnn_classifier.cpp:44]: I  think it doesn't need to be a private variable.

File: perception/traffic_light_classifier/src/cnn_classifier.cpp [hunk 2]
```diff
@@ -39,10 +39,14 @@ CNNClassifier::CNNClassifier(rclcpp::Node * node_ptr) : node_ptr_(node_ptr)
...
+  output_name_ = node_ptr_->declare_parameter("output_name", std::string("output_0"));
```
Comment (yukkysaito) [inline perception/traffic_light_classifier/src/cnn_classifier.cpp:45]: I think it doesn't need to be a private variable.
```suggestion
  input_name = node_ptr_->declare_parameter("input_name", std::string("input_0"));
  output_name = node_ptr_->declare_parameter("output_name", std::string("output_0"));
```

File: perception/traffic_light_classifier/src/cnn_classifier.cpp [hunk 3]
```diff
@@ -39,10 +39,14 @@ CNNClassifier::CNNClassifier(rclcpp::Node * node_ptr) : node_ptr_(node_ptr)
...
 
   readLabelfile(label_file_path, labels_);
 
-  trt_ = std::make_shared<Tn::TrtCommon>(model_file_path, precision);
+  trt_ = std::make_shared<Tn::TrtCommon>(model_file_path, precision, input_name_, output_name_);
```
Comment (yukkysaito) [inline perception/traffic_light_classifier/src/cnn_classifier.cpp:49]: related link : https://github.com/autowarefoundation/autoware.universe/pull/2794/files#r1093340555
```suggestion
  trt_ = std::make_shared<Tn::TrtCommon>(model_file_path, precision, input_name, output_name);
```

File: perception/traffic_light_classifier/include/traffic_light_classifier/cnn_classifier.hpp [hunk 4]
```diff
@@ -101,11 +101,13 @@ class CNNClassifier : public ClassifierInterface
...
   std::vector<std::string> labels_;
-  std::vector<float> mean_{0.242, 0.193, 0.201};
-  std::vector<float> std_{1.0, 1.0, 1.0};
+  std::vector<double> mean_;
+  std::vector<double> std_;
   int input_c_;
   int input_h_;
   int input_w_;
+  std::string input_name_;
+  std::string output_name_;
```
Comment (yukkysaito) [inline perception/traffic_light_classifier/include/traffic_light_classifier/cnn_classifier.hpp:110]: related link : https://github.com/autowarefoundation/autoware.universe/pull/2794/files#r1093340555
```suggestion
```

**Comment by yukkysaito** [comment]
Thank you for your PR 👍 I reviewed and commented

**Review by kaancolak — COMMENTED** [review]
LGTM

**Review by kaancolak — APPROVED** [review]

**Comment by mitsudome-r** [comment]
@yukkysaito Do you have another review? Otherwise, we would like to merge this once @Zeysthingz resolves the conflicts.

File: perception/traffic_light_classifier/launch/traffic_light_classifier.launch.xml [hunk 5]
```diff
@@ -4,20 +4,24 @@
...
     <param name="model_file_path" value="$(find-pkg-share traffic_light_classifier)/data/traffic_light_classifier_mobilenetv2.onnx"/>
     <param name="label_file_path" value="$(find-pkg-share traffic_light_classifier)/data/lamp_labels.txt"/>
+    <param name="mean" value="[0.242, 0.193, 0.201]"/>
+    <param name="std" value="[1.0, 1.0, 1.0]"/>
     <param name="precision" value="fp16"/>
+    <param name="input_name" value="input_0"/>
+    <param name="output_name" value="output_0"/>
     <param name="input_c" value="3"/>
     <param name="input_h" value="224"/>
     <param name="input_w" value="224"/>
```
Comment (yukkysaito) [inline perception/traffic_light_classifier/launch/traffic_light_classifier.launch.xml:27]: Could you write to README about this parameters?

Comment (Zeysthingz) [reply perception/traffic_light_classifier/launch/traffic_light_classifier.launch.xml:27]: Yes sure .

**Review by yukkysaito — APPROVED** [review]
LGTM

**Comment by kaancolak** [comment]
Hello @wep21 Daisuke-san,  Could you review this PR?

File: perception/traffic_light_classifier/utils/trt_common.cpp [hunk 6]
```diff
@@ -41,8 +41,10 @@ TrtCommon::TrtCommon(
   std::string model_path, std::string precision, std::string input_name, std::string output_name)
 : model_file_path_(model_path),
   precision_(precision),
+
   input_name_(input_name),
   output_name_(output_name),
+
```
Comment (wep21) [inline perception/traffic_light_classifier/utils/trt_common.cpp:47]: Are these two empty lines necessary?

**Comment by wep21** [comment]
Could you resolve some conflicts?

File: perception/traffic_light_classifier/utils/trt_common.cpp (hunk 6 above)
Comment (Zeysthingz) [reply perception/traffic_light_classifier/utils/trt_common.cpp:47]: No they are not

**Review by wep21 — APPROVED** [review]
LGTM

**Review by kaancolak — APPROVED** [review]

**Pull request merged**